src/coreComponents/mainInterface/initialization.cpp

```diff
@@ -253,12 +263,112 @@ std::unique_ptr< CommandLineOptions > parseCommandLineOptions( int argc, char *
 }
 
 
+void outputVersionInfo()
+{
+#if defined(__clang__)
+  GEOSX_LOG_RANK_0( "  - clang version: " <<
+                    __clang_major__<<"."<<
+                    __clang_minor__<<"."<<
+                    __clang_patchlevel__ );
+#if defined(__apple_build_version__)
+  GEOSX_LOG_RANK_0( "    apple clang version: " <<
+                    __apple_build_version__ );
+#endif
+#if defined(__ibmxl_vrm__)
+  GEOSX_LOG_RANK_0( "  - IBM compiler version: " <<
+                    __ibmxl_version__<<"."<<
+                    __ibmxl_release__<<"."<<
+                    __ibmxl_modification__<<"."<<
+                    __ibmxl_ptf_fix_level__ );
+#endif
+#elif defined(__GNUC__)
+  GEOSX_LOG_RANK_0( "  - gcc version: " <<
+                    __GNUC__<<"."<<
+                    __GNUC_MINOR__<<"."<<
+                    __GNUC_PATCHLEVEL__ );
+#endif
```

Comment:
Yeah...that is probably a better idea. Correct me if I am wrong, but `getCompilerIdString()` would just have a bunch of guarded statements like it is implemented, but instead of the `GEOSX_LOG_RANK_0()`, they would just fill the return string. Is that what you had in mind?

Reply:
Yes, exactly what I was thinking of

Reply:
I shouldn't be so lazy...not that I have time for this sort of "work".